{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Leukemia is the most common malignancy and the second most frequent general cause of childhood death. It is classified as acute lymphoid of the B- or T-lineage being the most prevalent type in children, and as acute myeloid. Although tremendous improvements have been made in the treatment of leukemia in the past few years, there is still a large proportion of patients who do not benefit from the available therapy. The overall survival of pediatric acute lymhoblastic leukemia (ALL) and acute myeloid leukemia (AML) patients treated with the current chemotherapy regimens is above 80 and 70%, respectively"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Idea behind the operation(Top 250 data  from GEO2R database which is microarray data  with Accession No GSE4072,  and the platform GPL2695 GEO2R Data(df) :https://www.ncbi.nlm.nih.gov/geo/geo2r/backend/geo2r.cgiand the genes which are  differentially expressed which are about 133 out of 250 rows are differentially expressed  gene_symbol('gs').This gene symbols are fed to the biological tool called STRING:(https://string-db.org/) to find a protein protein interaction this string gives a .tsv file which intern is input to the tool cytoscape cytoscape produces a data file which is named as cytoscape_data1 in the code)\n",
    "from this cytoscape data a new dataframe is created as df1.then top 10 from cytoscape data are selected as 'b' list\n",
    " and 'gs'as 'b' list bothe are compared to find match and i got 2 genes out of 133 in top 10 which are important to study about the lukemia cancer."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd   #simply import pandas \n",
    "import numpy as np\n",
    "#read dataset\n",
    "df=pd.read_excel(r'E:\\CSIR\\GEO2R Data.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pd alias of pandas \n",
    "#df name of dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>Gene.symbol</th>\n",
       "      <th>Gene.title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28659</td>\n",
       "      <td>0.419</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>-8.17</td>\n",
       "      <td>-3.82</td>\n",
       "      <td>-3.220</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37828</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>-7.81</td>\n",
       "      <td>-3.96</td>\n",
       "      <td>-3.812</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37010</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>-7.42</td>\n",
       "      <td>-3.97</td>\n",
       "      <td>-2.418</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>43508</td>\n",
       "      <td>0.431</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>-7.42</td>\n",
       "      <td>-4.17</td>\n",
       "      <td>-4.061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22218</td>\n",
       "      <td>0.476</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>-6.53</td>\n",
       "      <td>-3.89</td>\n",
       "      <td>-2.338</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>36232</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.013829</td>\n",
       "      <td>2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.853</td>\n",
       "      <td>PHC2</td>\n",
       "      <td>polyhomeotic homolog 2 (Drosophila)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>38008</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.013854</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>1.597</td>\n",
       "      <td>FLJ12529</td>\n",
       "      <td>pre-mRNA cleavage factor I, 59 kDa subunit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>28303</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.013950</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>1.158</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>24730</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.013954</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>38510</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.014039</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.832</td>\n",
       "      <td>MBP</td>\n",
       "      <td>myelin basic protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>250 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID  adj.P.Val   P.Value     t     B  logFC Gene.symbol  \\\n",
       "0    28659      0.419  0.000009 -8.17 -3.82 -3.220         NaN   \n",
       "1    37828      0.431  0.000026 -7.81 -3.96 -3.812         NaN   \n",
       "2    37010      0.431  0.000039 -7.42 -3.97 -2.418         NaN   \n",
       "3    43508      0.431  0.000039 -7.42 -4.17 -4.061         NaN   \n",
       "4    22218      0.476  0.000065 -6.53 -3.89 -2.338         NaN   \n",
       "..     ...        ...       ...   ...   ...    ...         ...   \n",
       "245  36232      1.000  0.013829  2.98 -4.24  0.853        PHC2   \n",
       "246  38008      1.000  0.013854  2.97 -4.24  1.597    FLJ12529   \n",
       "247  28303      1.000  0.013950  2.97 -4.24  1.158         NaN   \n",
       "248  24730      1.000  0.013954  2.97 -4.24  0.779         NaN   \n",
       "249  38510      1.000  0.014039  2.97 -4.24  0.832         MBP   \n",
       "\n",
       "                                     Gene.title  \n",
       "0                                           NaN  \n",
       "1                                           NaN  \n",
       "2                                           NaN  \n",
       "3                                           NaN  \n",
       "4                                           NaN  \n",
       "..                                          ...  \n",
       "245         polyhomeotic homolog 2 (Drosophila)  \n",
       "246  pre-mRNA cleavage factor I, 59 kDa subunit  \n",
       "247                                         NaN  \n",
       "248                                         NaN  \n",
       "249                        myelin basic protein  \n",
       "\n",
       "[250 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#datframe collection of rows and column\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gene_symbol=df['Gene.symbol'] #selected particular column which i needed this is nothing but the series with same datatype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.series.Series"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#see the type of the Gene_symbol\n",
    "type(Gene_symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>Gene.symbol</th>\n",
       "      <th>Gene.title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>33379</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>5.29</td>\n",
       "      <td>-3.97</td>\n",
       "      <td>1.520</td>\n",
       "      <td>PPP1R16B</td>\n",
       "      <td>protein phosphatase 1, regulatory (inhibitor) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36322</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>4.99</td>\n",
       "      <td>-3.99</td>\n",
       "      <td>1.555</td>\n",
       "      <td>C20orf77</td>\n",
       "      <td>chromosome 20 open reading frame 77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>34718</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>-4.83</td>\n",
       "      <td>-4.00</td>\n",
       "      <td>-1.335</td>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>activin A receptor type II-like 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>17745</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000686</td>\n",
       "      <td>4.82</td>\n",
       "      <td>-4.00</td>\n",
       "      <td>1.964</td>\n",
       "      <td>LGALS14</td>\n",
       "      <td>lectin, galactoside-binding, soluble, 14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>28554</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.001062</td>\n",
       "      <td>-4.98</td>\n",
       "      <td>-4.20</td>\n",
       "      <td>-2.878</td>\n",
       "      <td>TRIM31</td>\n",
       "      <td>tripartite motif-containing 31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>4784</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013692</td>\n",
       "      <td>-2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>-1.476</td>\n",
       "      <td>HPSE</td>\n",
       "      <td>heparanase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>27747</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013696</td>\n",
       "      <td>2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.850</td>\n",
       "      <td>ZNF552</td>\n",
       "      <td>zinc finger protein 552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>36232</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013829</td>\n",
       "      <td>2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.853</td>\n",
       "      <td>PHC2</td>\n",
       "      <td>polyhomeotic homolog 2 (Drosophila)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>38008</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013854</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>1.597</td>\n",
       "      <td>FLJ12529</td>\n",
       "      <td>pre-mRNA cleavage factor I, 59 kDa subunit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>38510</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.014039</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.832</td>\n",
       "      <td>MBP</td>\n",
       "      <td>myelin basic protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID  adj.P.Val   P.Value     t     B  logFC Gene.symbol  \\\n",
       "7    33379        1.0  0.000345  5.29 -3.97  1.520    PPP1R16B   \n",
       "9    36322        1.0  0.000534  4.99 -3.99  1.555    C20orf77   \n",
       "11   34718        1.0  0.000678 -4.83 -4.00 -1.335      ACVRL1   \n",
       "12   17745        1.0  0.000686  4.82 -4.00  1.964     LGALS14   \n",
       "15   28554        1.0  0.001062 -4.98 -4.20 -2.878      TRIM31   \n",
       "..     ...        ...       ...   ...   ...    ...         ...   \n",
       "243   4784        1.0  0.013692 -2.98 -4.24 -1.476        HPSE   \n",
       "244  27747        1.0  0.013696  2.98 -4.24  0.850      ZNF552   \n",
       "245  36232        1.0  0.013829  2.98 -4.24  0.853        PHC2   \n",
       "246  38008        1.0  0.013854  2.97 -4.24  1.597    FLJ12529   \n",
       "249  38510        1.0  0.014039  2.97 -4.24  0.832         MBP   \n",
       "\n",
       "                                            Gene.title  \n",
       "7    protein phosphatase 1, regulatory (inhibitor) ...  \n",
       "9                  chromosome 20 open reading frame 77  \n",
       "11                   activin A receptor type II-like 1  \n",
       "12            lectin, galactoside-binding, soluble, 14  \n",
       "15                      tripartite motif-containing 31  \n",
       "..                                                 ...  \n",
       "243                                         heparanase  \n",
       "244                            zinc finger protein 552  \n",
       "245                polyhomeotic homolog 2 (Drosophila)  \n",
       "246         pre-mRNA cleavage factor I, 59 kDa subunit  \n",
       "249                               myelin basic protein  \n",
       "\n",
       "[133 rows x 8 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#simple opetaion to drop the values which are Nan\n",
    "\n",
    "df.dropna(inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'adj.P.Val', 'P.Value', 't', 'B', 'logFC', 'Gene.symbol',\n",
       "       'Gene.title'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns    #....names of columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "133"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#using pandas we can count the no of entries in particular column or dataset \n",
    "\n",
    "(Gene_symbol).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#i reassigned the Gene_symbol to gs for simple access \n",
    "gs=(Gene_symbol).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "sort_by_pvalues=np.sort(['P.Values']) #........................sorted the dataframe by column P.Value in ascending order"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "sort_by_pvalues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>Gene.symbol</th>\n",
       "      <th>Gene.title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>33379</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>5.29</td>\n",
       "      <td>-3.97</td>\n",
       "      <td>1.520</td>\n",
       "      <td>PPP1R16B</td>\n",
       "      <td>protein phosphatase 1, regulatory (inhibitor) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36322</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>4.99</td>\n",
       "      <td>-3.99</td>\n",
       "      <td>1.555</td>\n",
       "      <td>C20orf77</td>\n",
       "      <td>chromosome 20 open reading frame 77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>34718</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>-4.83</td>\n",
       "      <td>-4.00</td>\n",
       "      <td>-1.335</td>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>activin A receptor type II-like 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>17745</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000686</td>\n",
       "      <td>4.82</td>\n",
       "      <td>-4.00</td>\n",
       "      <td>1.964</td>\n",
       "      <td>LGALS14</td>\n",
       "      <td>lectin, galactoside-binding, soluble, 14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>28554</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.001062</td>\n",
       "      <td>-4.98</td>\n",
       "      <td>-4.20</td>\n",
       "      <td>-2.878</td>\n",
       "      <td>TRIM31</td>\n",
       "      <td>tripartite motif-containing 31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>4784</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013692</td>\n",
       "      <td>-2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>-1.476</td>\n",
       "      <td>HPSE</td>\n",
       "      <td>heparanase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>27747</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013696</td>\n",
       "      <td>2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.850</td>\n",
       "      <td>ZNF552</td>\n",
       "      <td>zinc finger protein 552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>36232</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013829</td>\n",
       "      <td>2.98</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.853</td>\n",
       "      <td>PHC2</td>\n",
       "      <td>polyhomeotic homolog 2 (Drosophila)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>38008</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.013854</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>1.597</td>\n",
       "      <td>FLJ12529</td>\n",
       "      <td>pre-mRNA cleavage factor I, 59 kDa subunit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>38510</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.014039</td>\n",
       "      <td>2.97</td>\n",
       "      <td>-4.24</td>\n",
       "      <td>0.832</td>\n",
       "      <td>MBP</td>\n",
       "      <td>myelin basic protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID  adj.P.Val   P.Value     t     B  logFC Gene.symbol  \\\n",
       "7    33379        1.0  0.000345  5.29 -3.97  1.520    PPP1R16B   \n",
       "9    36322        1.0  0.000534  4.99 -3.99  1.555    C20orf77   \n",
       "11   34718        1.0  0.000678 -4.83 -4.00 -1.335      ACVRL1   \n",
       "12   17745        1.0  0.000686  4.82 -4.00  1.964     LGALS14   \n",
       "15   28554        1.0  0.001062 -4.98 -4.20 -2.878      TRIM31   \n",
       "..     ...        ...       ...   ...   ...    ...         ...   \n",
       "243   4784        1.0  0.013692 -2.98 -4.24 -1.476        HPSE   \n",
       "244  27747        1.0  0.013696  2.98 -4.24  0.850      ZNF552   \n",
       "245  36232        1.0  0.013829  2.98 -4.24  0.853        PHC2   \n",
       "246  38008        1.0  0.013854  2.97 -4.24  1.597    FLJ12529   \n",
       "249  38510        1.0  0.014039  2.97 -4.24  0.832         MBP   \n",
       "\n",
       "                                            Gene.title  \n",
       "7    protein phosphatase 1, regulatory (inhibitor) ...  \n",
       "9                  chromosome 20 open reading frame 77  \n",
       "11                   activin A receptor type II-like 1  \n",
       "12            lectin, galactoside-binding, soluble, 14  \n",
       "15                      tripartite motif-containing 31  \n",
       "..                                                 ...  \n",
       "243                                         heparanase  \n",
       "244                            zinc finger protein 552  \n",
       "245                polyhomeotic homolog 2 (Drosophila)  \n",
       "246         pre-mRNA cleavage factor I, 59 kDa subunit  \n",
       "249                               myelin basic protein  \n",
       "\n",
       "[133 rows x 8 columns]"
      ]
     },
     "execution_count": 331,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [],
   "source": [
    " #....exported new file  CYTOSCAPE FILE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "cytoscape_data=pd.read_excel(r'C:\\Users\\dell\\Desktop\\cytoscape_data1.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "cytoscape_sorted=np.sort(['Degree'])#............sorted by degree descenind "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>BetweennessCentrality</th>\n",
       "      <th>ClosenessCentrality</th>\n",
       "      <th>NeighborhoodConnectivity</th>\n",
       "      <th>Degree</th>\n",
       "      <th>NumberOfUndirectedEdges</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>0.116004</td>\n",
       "      <td>0.491582</td>\n",
       "      <td>17.631579</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HDAC2</td>\n",
       "      <td>0.077789</td>\n",
       "      <td>0.467949</td>\n",
       "      <td>14.972973</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRC</td>\n",
       "      <td>0.035786</td>\n",
       "      <td>0.437126</td>\n",
       "      <td>18.515152</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>0.036963</td>\n",
       "      <td>0.433234</td>\n",
       "      <td>18.600000</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>0.039844</td>\n",
       "      <td>0.445122</td>\n",
       "      <td>19.034483</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>ARHGAP33</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>ARHGAP23</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>DCAF4</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.266910</td>\n",
       "      <td>9.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>TET3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.304167</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>CD82</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.308668</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         name  BetweennessCentrality  ClosenessCentrality  \\\n",
       "1        AKT1               0.116004             0.491582   \n",
       "2       HDAC2               0.077789             0.467949   \n",
       "3         SRC               0.035786             0.437126   \n",
       "4        HRAS               0.036963             0.433234   \n",
       "5        EGFR               0.039844             0.445122   \n",
       "..        ...                    ...                  ...   \n",
       "148  ARHGAP33               0.000000             0.290837   \n",
       "149  ARHGAP23               0.000000             0.290837   \n",
       "150     DCAF4               0.000000             0.266910   \n",
       "151      TET3               0.000000             0.304167   \n",
       "152      CD82               0.000000             0.308668   \n",
       "\n",
       "     NeighborhoodConnectivity  Degree  NumberOfUndirectedEdges  \n",
       "1                   17.631579      38                       38  \n",
       "2                   14.972973      37                       37  \n",
       "3                   18.515152      33                       33  \n",
       "4                   18.600000      30                       30  \n",
       "5                   19.034483      29                       29  \n",
       "..                        ...     ...                      ...  \n",
       "148                 22.000000       1                        1  \n",
       "149                 22.000000       1                        1  \n",
       "150                  9.000000       1                        1  \n",
       "151                 22.000000       1                        1  \n",
       "152                 29.000000       1                        1  \n",
       "\n",
       "[152 rows x 6 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cytoscape_data.iloc[1::]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "42"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cytoscape_data['Degree'].max() #find highest value in the column using max() function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "cytoscape_sorted=np.sort(['Degree'], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cytoscape_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "cytoscape_data=pd.DataFrame(cytoscape_data,columns=['name','Degree','BetweennessCentrality','ClosenessCentrality','NeighborhoodConnectivity'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>Degree</th>\n",
       "      <th>BetweennessCentrality</th>\n",
       "      <th>ClosenessCentrality</th>\n",
       "      <th>NeighborhoodConnectivity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HDAC1</td>\n",
       "      <td>42</td>\n",
       "      <td>0.115025</td>\n",
       "      <td>0.480263</td>\n",
       "      <td>14.523810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>38</td>\n",
       "      <td>0.116004</td>\n",
       "      <td>0.491582</td>\n",
       "      <td>17.631579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HDAC2</td>\n",
       "      <td>37</td>\n",
       "      <td>0.077789</td>\n",
       "      <td>0.467949</td>\n",
       "      <td>14.972973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRC</td>\n",
       "      <td>33</td>\n",
       "      <td>0.035786</td>\n",
       "      <td>0.437126</td>\n",
       "      <td>18.515152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>30</td>\n",
       "      <td>0.036963</td>\n",
       "      <td>0.433234</td>\n",
       "      <td>18.600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>ARHGAP33</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>ARHGAP23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>DCAF4</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.266910</td>\n",
       "      <td>9.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>TET3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.304167</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>CD82</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.308668</td>\n",
       "      <td>29.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         name  Degree  BetweennessCentrality  ClosenessCentrality  \\\n",
       "0       HDAC1      42               0.115025             0.480263   \n",
       "1        AKT1      38               0.116004             0.491582   \n",
       "2       HDAC2      37               0.077789             0.467949   \n",
       "3         SRC      33               0.035786             0.437126   \n",
       "4        HRAS      30               0.036963             0.433234   \n",
       "..        ...     ...                    ...                  ...   \n",
       "148  ARHGAP33       1               0.000000             0.290837   \n",
       "149  ARHGAP23       1               0.000000             0.290837   \n",
       "150     DCAF4       1               0.000000             0.266910   \n",
       "151      TET3       1               0.000000             0.304167   \n",
       "152      CD82       1               0.000000             0.308668   \n",
       "\n",
       "     NeighborhoodConnectivity  \n",
       "0                   14.523810  \n",
       "1                   17.631579  \n",
       "2                   14.972973  \n",
       "3                   18.515152  \n",
       "4                   18.600000  \n",
       "..                        ...  \n",
       "148                 22.000000  \n",
       "149                 22.000000  \n",
       "150                  9.000000  \n",
       "151                 22.000000  \n",
       "152                 29.000000  \n",
       "\n",
       "[153 rows x 5 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cytoscape_data    #..created new dataframe with five columns only which are the strong features of the gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1=cytoscape_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>Degree</th>\n",
       "      <th>BetweennessCentrality</th>\n",
       "      <th>ClosenessCentrality</th>\n",
       "      <th>NeighborhoodConnectivity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HDAC1</td>\n",
       "      <td>42</td>\n",
       "      <td>0.115025</td>\n",
       "      <td>0.480263</td>\n",
       "      <td>14.523810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>38</td>\n",
       "      <td>0.116004</td>\n",
       "      <td>0.491582</td>\n",
       "      <td>17.631579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HDAC2</td>\n",
       "      <td>37</td>\n",
       "      <td>0.077789</td>\n",
       "      <td>0.467949</td>\n",
       "      <td>14.972973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRC</td>\n",
       "      <td>33</td>\n",
       "      <td>0.035786</td>\n",
       "      <td>0.437126</td>\n",
       "      <td>18.515152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>30</td>\n",
       "      <td>0.036963</td>\n",
       "      <td>0.433234</td>\n",
       "      <td>18.600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>ARHGAP33</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>ARHGAP23</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.290837</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>DCAF4</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.266910</td>\n",
       "      <td>9.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>TET3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.304167</td>\n",
       "      <td>22.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>CD82</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.308668</td>\n",
       "      <td>29.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         name  Degree  BetweennessCentrality  ClosenessCentrality  \\\n",
       "0       HDAC1      42               0.115025             0.480263   \n",
       "1        AKT1      38               0.116004             0.491582   \n",
       "2       HDAC2      37               0.077789             0.467949   \n",
       "3         SRC      33               0.035786             0.437126   \n",
       "4        HRAS      30               0.036963             0.433234   \n",
       "..        ...     ...                    ...                  ...   \n",
       "148  ARHGAP33       1               0.000000             0.290837   \n",
       "149  ARHGAP23       1               0.000000             0.290837   \n",
       "150     DCAF4       1               0.000000             0.266910   \n",
       "151      TET3       1               0.000000             0.304167   \n",
       "152      CD82       1               0.000000             0.308668   \n",
       "\n",
       "     NeighborhoodConnectivity  \n",
       "0                   14.523810  \n",
       "1                   17.631579  \n",
       "2                   14.972973  \n",
       "3                   18.515152  \n",
       "4                   18.600000  \n",
       "..                        ...  \n",
       "148                 22.000000  \n",
       "149                 22.000000  \n",
       "150                  9.000000  \n",
       "151                 22.000000  \n",
       "152                 29.000000  \n",
       "\n",
       "[153 rows x 5 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.dropna()           #.......again data is cleaned "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# top 10 values are selected for the comparision \n",
    "b=df1.loc[0:10].tolist()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found the match at 124th position\n",
      "found the match at 28th position\n"
     ]
    }
   ],
   "source": [
    "for i in range(0,10):\n",
    "    \n",
    "    for j in range(0,133): \n",
    "        \n",
    "        if a[j]==b[i]:       \n",
    "             print(\"found the match at {}th position\".format(j))\n",
    "        \n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
